cpp/src/arrow/dataset/projector.cc

```diff
@@ -86,32 +113,23 @@ Result<std::shared_ptr<RecordBatch>> RecordBatchProjector::Project(
 
 Status RecordBatchProjector::SetInputSchema(std::shared_ptr<Schema> from,
                                             MemoryPool* pool) {
+  RETURN_NOT_OK(CheckProjectable(*from, *to_));
```

Comment:
If I read this diff correctly, the only place that the R package tests should touch new code (i.e. without calling ReplaceSchema, which has a binding added but no test) is here. Is there something suspicious about CheckProjectable (on 32-bit mingw)?